### ISSUE TITLE
check folder permission before make install? Enforce prefix?

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cmake-buildscripts#48]
1. saying I do

`cmake ../amber`
`make install` -> Got stuck in this step due to no permission to write to default install folder.
How's about checking permission?

2. How's about enforce proving prefix rather using the system folder?

**Comment by multiplemonomials** [comment]
> How's about checking permission?
I can do that, sure.

> How's about enforce proving prefix rather using the system folder?
I really don't understand what you're asking here.  Can you rephrase?  Thanks

**Comment by hainm** [comment]
> I really don't understand what you're asking here. Can you rephrase? Thanks

Not using default install folder (Just bring here for discussion).

**Comment by multiplemonomials** [comment]
So, you mean require the user to set `CMAKE_INSTALL_PREFIX` instead of defaulting it?

**Comment by hainm** [comment]
yeah (just want to hear your idea about this).

**Comment by hainm** [comment]
btw, what's the default folder? I have hard time find it. :D

**Comment by multiplemonomials** [comment]
You have a point that it would be bad if a user didn't know to set `CMAKE_INSTALL_PREFIX`, and Amber got installed somewhere they didn't want it.  I'm hoping that the
```
Install location:      /opt/amber
```
line in the build report will help with this.

Currently, the default `CMAKE_INSTALL_PREFIX` comes from CMake.  On Linux (and maybe mac???) it's `/usr/local`.  On Windows, it's `C:\Program Files\Amber`.

**Comment by multiplemonomials** [comment]
I could override this from my scripts.  I was thinking:

| OS     | Prefix |
| ------- | ---------- | 
| Linux | `/opt/amber` | 
| Mac | `/Applications/Amber MD` |
| Windows | `C:\Program Files\Amber` |

**Comment by hainm** [comment]
cc @dacase, @swails and @drroe what do you think?

**Comment by multiplemonomials** [comment]
OK I added the check that the install prefix is writable in 52c7421273
